io/include/detray/io/common/geometry_writer.hpp

```diff
@@ -97,43 +101,9 @@ class geometry_writer : public writer_interface<detector_t> {
               std::enable_if_t<!std::is_same_v<typename mask_t::shape, void>,
                                bool> = true>
     static mask_payload serialize(const mask_t& m) {
-        using shape_id = mask_payload::mask_shape;
         mask_payload mask_data;
 
-        // Find the correct shape index (use name for simplicity)
-        using shape_t = typename mask_t::shape;
-        if constexpr (std::is_same_v<shape_t, rectangle2D<>>) {
-            mask_data.shape = shape_id::rectangle2;
-        } else if constexpr (std::is_same_v<shape_t, trapezoid2D<>>) {
-            mask_data.shape = shape_id::trapezoid2;
-        } else if constexpr (std::is_same_v<shape_t, cylinder2D<>>) {
-            mask_data.shape = shape_id::cylinder2;
-        } else if constexpr (std::is_same_v<
-                                 shape_t,
-                                 cylinder2D<false,
-                                            cylinder_portal_intersector>>) {
-            mask_data.shape = shape_id::portal_cylinder2;
-        } else if constexpr (std::is_same_v<shape_t, ring2D<>>) {
-            mask_data.shape = shape_id::ring2;
-        } else if constexpr (std::is_same_v<shape_t, annulus2D<>>) {
-            mask_data.shape = shape_id::annulus2;
-        } else if constexpr (std::is_same_v<shape_t, line<true>>) {
-            mask_data.shape = shape_id::cell_wire;
-        } else if constexpr (std::is_same_v<shape_t, line<false>>) {
-            mask_data.shape = shape_id::straw_wire;
-        } else if constexpr (std::is_same_v<shape_t, single3D<0>>) {
-            mask_data.shape = shape_id::single1;
-        } else if constexpr (std::is_same_v<shape_t, single3D<1>>) {
-            mask_data.shape = shape_id::single2;
-        } else if constexpr (std::is_same_v<shape_t, single3D<2>>) {
-            mask_data.shape = shape_id::single3;
-        } else if constexpr (std::is_same_v<shape_t, cuboid3D<>>) {
-            mask_data.shape = shape_id::cuboid3;
-        } else if constexpr (std::is_same_v<shape_t, cylinder3D>) {
-            mask_data.shape = shape_id::cylinder3;
-        } else {
-            mask_data.shape = shape_id::unknown;
-        }
```

Comment:
This removal is not really related to the grid PR, right?

Reply:
Correct, I just re-arranged this to keep the code structure the same between the writers.